metricbeat/module/kubernetes/state_container/state_container.go

```diff
@@ -130,18 +142,29 @@ func (m *MetricSet) Fetch(reporter mb.ReporterV2) error {
 	}
 
 	m.enricher.Enrich(events)
-
+	//m.Logger().Infof("Events are %+v", events)
 	// Calculate deprecated nanocores values
 	for _, event := range events {
-		if request, ok := event["cpu.request.cores"]; ok {
-			if requestCores, ok := request.(float64); ok {
-				event["cpu.request.nanocores"] = requestCores * nanocores
-			}
-		}
-
-		if limit, ok := event["cpu.limit.cores"]; ok {
-			if limitCores, ok := limit.(float64); ok {
-				event["cpu.limit.nanocores"] = limitCores * nanocores
+		if cpuFields, ok := event["cpu"]; ok {
```

Comment:
Why not to convert the `event` to `common.MapStr` and then access the `cpu.limit.cores` field directly, so as to avoid all these `if`s? With the methods that `MapStr` provides you should be ok.

Reply:
Good catch. `event` is already a `common.MapStr`